File: app/src/main/java/br/com/ufg/www/events/mvp/login/Presenter.java
package br.com.ufg.www.events.mvp.login;

import br.com.ufg.www.events.model.Login;

public class Presenter implements Contract.Presenter {

    private Contract.View view;
    private Interactor interactor;

    public Presenter(Contract.View view) {
        this.view = view;
        this.interactor = new Interactor();
    }

    @Override
    public void login(Login login) {
        Boolean result = interactor.login(login);
        if (result) {
            view.onLoggedIn();
        } else {
            view.loginFailed();
        }
    }
}

File: app/src/main/java/br/com/ufg/www/events/mvp/user/login/Presenter.kt
package br.com.ufg.www.events.mvp.user.login

import br.com.ufg.www.events.App
import br.com.ufg.www.events.model.Login

class Presenter(private val view: Contract.View) : Contract.Presenter {

    private val interactor = Interactor()

    override fun login(login: Login) {
        val callback = { result: Boolean ->
            if (result) {
                App.userLoggedIn = login.login
                view.onLoggedIn()
            } else {
                view.loginFailed()
            }
        }

        interactor.login(login, callback).start()
    }
}
